include/picongpu/Perf.cpp

```diff
@@ -0,0 +1,57 @@
+// perf.cc
+#include "pmacc/eventSystem/Perf.hpp"
+#include <vector>
+#include <tuple>
+#include <map>
+#include <cmath>
+
+namespace Performance {
+    using d2 = std::tuple<double,double>;
+
+    static bool tracing = false;
+    static std::map<std::string,std::map<d2,PerfAvg>> events;
+
+    void Timers::append(const std::string &label, PerfData &datum) {
```

Comment:
only a note: Using a lambda function as a parameter that is stringifying your data, whatever data means` will give you the possibility to remove the full stringify overhead.

```C++
foo(auto lamda)
{
   if(tracingEnabled)
   {
      // lazy evaluation of the lambda only if tracing is enabled, this will reduce the overhead.
      std::tuple<string,string> data = lamda();
      mymap[get<0>(data] = get<1>(data);
   }
}
```

Reply:
We should consider this later.  The profiling code also needs a good on/off switch.